pkg/operator/v1helpers/args.go

```diff
@@ -0,0 +1,103 @@
+package v1helpers
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// shellEscapePattern determines if a string should be enclosed in single quotes
+// so that it can safely be passed to shell command line.
+var shellEscapePattern *regexp.Regexp
+
+func init() {
+	// some characters have special meaning to the shell and may need to be escaped
+	// the following regexp defines safe characters that don't require escaping
+	shellEscapePattern = regexp.MustCompile(`[^\w@%+=:,./-]`)
+}
+
+// FlagsFromUnstructuredEscaped process the given arguments and determines if they should be escaped so that they can be safely passed as shell command line arguments.
+// unstructuredArgs holds unprocessed arguments usually retrieved from an operator's configuration file under a specific key.
+//
+// Use ToFlagSlice function to get a slice of string flags.
+func FlagsFromUnstructuredEscaped(unstructuredArgs map[string]interface{}) (map[string][]string, error) {
```

Comment:
I still think this is the wrong spot for escaping. Escape when turning a `[]string` into *one* string.